flang/lib/Semantics/check-cuda.cpp

```diff
@@ -387,13 +387,11 @@ template <bool IsCUFKernelDo> class DeviceContextChecker {
               Check(x.value());
             },
             [&](const common::Indirection<parser::AssignmentStmt> &x) {
-              if (IsCUFKernelDo) {
-                const evaluate::Assignment *assign{
-                    semantics::GetAssignment(x.value())};
-                if (assign) {
-                  ErrorIfHostSymbol(assign->lhs, source);
-                  ErrorIfHostSymbol(assign->rhs, source);
-                }
+              const evaluate::Assignment *assign{
```

Comment:
`assign` should be local to the predicate of the `if` if it is not used thereafter.